src/cbmc/cbmc_parse_options.cpp

```diff
@@ -450,6 +457,27 @@ void cbmc_parse_optionst::get_command_line_options(optionst &options)
   PARSE_OPTIONS_GOTO_TRACE(cmdline, options);
 }
 
+void cbmc_parse_optionst::check_smt2(
+  const std::string &name,
+  const std::string &pre_message)
+{
+  int exit_code;
+#ifdef _WIN32
+  {
+    std::string where_string = "%%windir%%\\system32\\where.exe";
+    exit_code = run(where_string, {where_string, name}, "", "", "");
+  }
+#else
+  exit_code = run("which", {"which", name}, "", "", "");
+#endif
+  if(exit_code != 0)
+  {
+    log.error() << pre_message << "SMT2 solver \'" << name
+                << "\' is not on the PATH" << messaget::eom;
+    exit(CPROVER_EXIT_USAGE_ERROR);
```

Comment:
Instead of exiting here I think we should throw an exception instead and let the logging and exiting be handled in `parse_options_baset::main()`.